src/handlers/deleteCloudFormationStack.ts

```diff
@@ -1,76 +1,83 @@
+import { deleteStack } from "../cloudformation";
 
-const cloudFormation = new CloudFormation();
+class StackJanitorNotEnabledError extends Error {
+  constructor(message: string) {
+    super(message);
+  }
+}
 
-export enum Action {
-  REMOVE = "REMOVE"
+export class ValidationError extends Error {
+  constructor(message: string) {
+    super(message);
+  }
 }
 
-export const getStackNamesFromStreamEvent = (
-  event: DynamoDBStreamEvent
-): (string | undefined)[] => {
-  const removeRecords = event.Records.filter(
-    record => record.eventName === Action.REMOVE
-  );
+function isStackJanitorEnabled(tags: CustomTag[]) {
+  const status = tags.find(tag => tag.key === "stackjanitor")?.value;
+  return status === StackStatus.Enabled;
+}
 
-  return removeRecords.map((record: DynamoDBRecord) => {
-    if (record.dynamodb && record.dynamodb.Keys) {
-      return record.dynamodb.Keys.stackName.S;
-    }
-  });
-};
+export const deleteCloudFormationStack = async (item: DataItem) => {
+  const { stackName, tags } = item;
 
-export const checkStackJanitorStatus = async (
-  Stacks: CloudFormation.Stack[]
-) => {
-  try {
-    const tags = getTagsFromStacks(Stacks);
-    const customTags = convertTags(tags);
-    return getStackJanitorStatus(customTags);
-  } catch (e) {
-    logger.error(e);
-    return StackStatus.Disabled;
+  const isEnabled = isStackJanitorEnabled(JSON.parse(tags));
+
+  if (!isEnabled) {
+    throw new StackJanitorNotEnabledError(
+      `StackJanitor is not enabled for ${stackName}`
+    );
   }
+
+  logger.info(`Deleting CFN stack: ${stackName}`);
+  await deleteStack(stackName);
+  logger.info(`CFN Stack: ${stackName} deleted successfully`);
 };
 
-export const deleteCloudFormationStack = async (
-  event: DynamoDBStreamEvent,
-  cloudFormation: CloudFormation
-) => {
-  const StackNames = await getStackNamesFromStreamEvent(event);
+async function processRecords(records: ParsedRecord<DataItem>[]) {
+  for (const record of records) {
+    const { eventID, oldData, eventName } = record;
+    const eventDetails = `Event ID: ${eventID}, Event Name: ${eventName}`;
 
-  for (let StackName of StackNames) {
-    if (!StackName) {
+    logger.info(`Started processing Dynamo stream. ${eventDetails}`);
+
+    if (!(eventName === DynamoDBEventType.Remove) || !oldData) {
+      logger.info(`Ignoring event ${eventName}. ${eventDetails}`);
       return;
     }
 
-    const { Stacks } = await cloudFormation
-      .describeStacks({ StackName })
-      .promise();
-
-    if (!Stacks) {
+    if (!oldData) {
+      logger.info(`Data is null ${eventName}. Cannot proceed. ${eventDetails}`);
       return;
     }
 
-    const status = await checkStackJanitorStatus(Stacks);
-    if (status === StackStatus.Enabled) {
-      try {
-        await cloudFormation.deleteStack({ StackName }).promise();
-      } catch (e) {
-        logger.error(e);
+    try {
+      await deleteCloudFormationStack(oldData);
+    } catch (err) {
+      if (
```

Comment:
nit: Maybe no need to split the errors here since all of them simply call `logger.error(`${err.message} - ${eventDetails}`);`?